skbuild/cmaker.py

```diff
@@ -87,6 +87,34 @@ def get_cmake_version(cmake_executable=CMAKE_DEFAULT_EXECUTABLE):
     return version_string.splitlines()[0].split(' ')[-1]
 
 
+def env_without_destdir(env):
```

Comment:
Last, would be stellar to update the restructured text documentation to document this behavior.

This could be done adding a section `Environment` before the `Advanced Usage` section in [docs/usage.rst#advanced-usage](https://github.com/scikit-build/scikit-build/blob/master/docs/usage.rst#advanced-usage)

For now the section would simply list each variables and the effect of scikit-build on each:

```
* DESTDIR: ...

* MAKEFLAGS: ...
```

Reply:
Done.